### ISSUE TITLE
Video Player

### ISSUE THREAD

**Issue by dman926** [dman926/SEPHIRA#21]
**Is your feature request related to a problem? Please describe.**
A common request from clients is a video/live stream player.

**Describe the solution you'd like**
A video player module & component would be a big improvement for the ease of use of SEPHIRA.

**Describe alternatives you've considered**
I've used [Videogular2](https://videogular.github.io/videogular2/) before for a different project. It worked well enough and was easy to set up, but had some limitations, notably auto refreshing the player when a stream became available. [Video.js](https://videojs.com/) might be a good alternative. Need to try it out to see if it'll be better.

**Comment by dman926** [comment]
Video.js doesn't work very well for me. Its nice and simple, but it fails to load with a status_code=206 on the video stream endpoint in chrome, but works fine in Firefox, and will break in Firefox if the status_code=200 and works in chrome. Will try out ngx-videogular (videogular2 successor)

**Comment by dman926** [comment]
ngx-videogular works well. I fixed the streaming endpoint. Turns out the issue was a combination of the chunk size being too large and the `content-range` header overflowing the end byte variable. I don't know if I want to return to video.js or stick with ngx-videogular. video.js will be easier to componentize compared to videogular, but I know videogular works and is built for angular

**Comment by dman926** [comment]
Going to retry using video.js instead. ngx videogular looks nice out of the box, but it lacks documentation and is difficult to use the api, specifically for adding subtitle tracks

**Comment by dman926** [comment]
Yeah, video.js plays much nicer compared to ngx videogular. It works flawlessly (except for slow loading large files in chrome, but not firefox) streaming both video and subtitle tracks. The main problem with ngx videogular was that it refused to allow streaming of tracks. Merging with `angular-refresh`, but leaving this issue open in case I run into more issues (I'm sure I will)

**Comment by dman926** [comment]
Besides styling and livestream testing, the video player is working well. I moved the logic of adding and removing source/track tags for `ngx-videogular` and replaced it with a simple `addMedia(media: Media)` function that takes in a `Media` object and will add sources, tracks, and posters where for the media and associated media.
I also created a metadata field for Media to store extra info such as a vtt subtitles's language and display name
The player is a little slow to load on chrome, but that seems to a chrome issue since it loads quick on firefox.
I'm closing this issue for now, and will probably make another issue for styling stuff if I need to in the future.
